### DISCUSSION TITLE
Looking for advice on building the smallest tidb cluster

### DISCUSSION THREAD

**Discussion by laoba25** [pingcap/tidb#40842] [Q&A]
---
Hi community:

Happy New Year!

Due to the needs of project development, I recently wanted to set up a tidb cluster.
Through the official website, I learned that the minimum machine size required by tidb is as many as 10.
However, due to the limited project budget, only a maximum of 6 machines are accepted.

Therefore, I want to know whether it is possible to reduce the number of cluster machines as much as possible. Specifically, there are two questions as follows.

Is it possible to use 6 machines with medium and high configuration (32 physical core + 128 GB) to complete the deployment of all core components
The topology is as follows:
1. 3 machines: 
   Mixed deployment of PD and TiDB
2. The other 3 machines
   mixed deployment TiKV, TiFlash
Whether to support tiup deployment + k8s deployment mix

For example:
Monitoring & Grafana and TiCDC clusters are expected to be deployed in independent k8s clusters to avoid occupying 6 physical machine resources.

Other components are deployed directly on the physical machine using tiup. I wonder if this deployment method is supported?

Any advice is greatly appreciated. 🙏

---
hi 社区：

过年好！

因为项目发展的需要，最近想着手组建 tidb 集群。
通过官网了解到 tidb 所需最小的机器规模多达10台。
但因项目预算受限，仅最多接受6台机器规模。
所以想了解，有没有可能将集群机器数量尽量压缩，具体有如下两个问题。

1. 是否可以使用6台中高配机器（32 物理core + 128 GB）完成全部核心组件部署
拓扑如下：
- 3台机器 ：混合部署 PD 、TiDB
- 另3台机器：混合部署 TiKV、TiFlash

2. 是否支持 tiup 部署 + k8s 部署混用

例如：
Monitoring & Grafana 以及 TiCDC 集群，预想部署于独立的 k8s 集群，避免占用6台物理机资源。
其他组件均采用 tiup 直接在物理机上部署，不知道是否支持此部署方法？

如果能得到任何建议，都非常感谢。🙏

**Comment by zhangyangyu** [comment]
Hello, I would recommend asking this question in [askTUG](https://asktug.com/) for Chinese or https://ask.pingcap.com for English. Usually people in TiDB community use these platforms as forum instead of GitHub discussion.
